account_cancel_with_reversal/tests/test_account_move_reverse.py

```diff
@@ -0,0 +1,131 @@
+# -*- coding: utf-8 -*-
+from datetime import date
+from openerp.tests import common
+
+
+class TestAccountInvoiceCancel(common.TransactionCase):
+    """
+    Case1 - test_invoice_cancel:
+    - Create draft invoice and validate and Cancel
+    - When invoice cancel reverse entry should be generated.
+    Case2 - test_voucher_cancel:
+    - Create draft voucher and validate and Unreconcile
+    - When voucher Unreconcile reverse entry should be generated.
+    """
+    def setUp(self):
+        super(TestAccountInvoiceCancel, self).setUp()
+        cr, uid = self.cr, self.uid
+        self.account_invoice_model = self.registry('account.invoice')
+        self.account_invoice_line_model = self.registry('account.invoice.line')
+        self.wizard_voucher_move_reverse = \
+            self.registry('account.move.reverse')
+        self.account_voucher_model = self.registry('account.voucher')
+        self.model_data = self.registry("ir.model.data")
+
+        self.partner_id = self.model_data.get_object_reference(
+            cr, uid, "base", "res_partner_2")[1]
+        self.account_id = self.model_data.get_object_reference(
+            cr, uid, "account", "a_recv")[1]
+        self.product_id = self.model_data.get_object_reference(
+            cr, uid, "product", "product_product_6")[1]
+        self.period_id = self.model_data.get_object_reference(
+            cr, uid, "account", "period_10")[1]
+        self.journal_id = self.model_data.get_object_reference(
+            cr, uid, "account", "bank_journal")[1]
+        self.currency_id = self.model_data.get_object_reference(
+            cr, uid, "base", "EUR")[1]
+        self.company_id = self.model_data.get_object_reference(
+            cr, uid, "base", "main_company")[1]
+
+    def test_invoice_cancel(self):
+        cr, uid = self.cr, self.uid
+
+        invoice_line_val = {
+            'name': 'LCD Screen',
+            'product_id': self.product_id,
+            'quantity': 5,
+            'price_unit': 200,
+        }
+
+        self.invoice_id = self.account_invoice_model.create(
+            cr, uid, {'partner_id': self.partner_id,
+                      'account_id': self.account_id,
+                      'journal_id': self.journal_id,
+                      'company_id': self.company_id,
+                      'invoice_line': [(0, 0, invoice_line_val)]})
+        invoice = self.account_invoice_model.browse(cr, uid, self.invoice_id)
+        self.account_invoice_model.signal_workflow(
+            cr, uid, [self.invoice_id], 'invoice_open')
+        self.assertEqual(invoice.state, 'open', 'Invoice not validated.')
+        move_id = invoice.move_id
+
+        reverse_move_id = move_id.create_reversals(
```

Comment:
In testing, we should not call `create_reversals` directly. I saw method `action_reverse_invoice()` in wizard. Should we call this, and the reversed entry should be created?

In this test, I see you are creating reverse once, and then trigger workflow invoice_cancel again. Then, it is not testing the addon.